src/components/VmActions/index.js

```diff
@@ -134,15 +136,40 @@ class VmActions extends React.Component {
 
     const vncConsole = vm.get('consoles').find(c => c.get('protocol') === 'vnc')
     const hasRdp = isWindows(vm.getIn(['os', 'type']))
-    const consoles = vm.get('consoles').map((c) => ({
-      priority: 0,
-      shortTitle: msg[c.get('protocol') + 'Console'](),
-      icon: <Icon name='external-link' />,
-      id: `${idPrefix}-button-console-${c.get('protocol')}`,
-      confirmation: (
-        <ConsoleConfirmationModal consoleId={c.get('id')} vm={vm} />
-      ),
-    })).toJS()
+    let consoles = []
+    if (isOnCard) {
+      consoles.push({
+        priority: 1,
+        actionDisabled: isPool || !canConsole(status) || vm.getIn(['actionInProgress', 'getConsole']),
+        shortTitle: msg.console(),
+        onClick: (e) => { e.preventDefault() },
+        className: 'btn btn-success',
+        id: `${idPrefix}-button-Consoles`,
+        confirmation: (<AutoSelectConsole vm={vm} isNoVNC id={`${vm.get('id')}-console-selector`} />),
+      })
+    } else {
+      consoles = vm.get('consoles').map((c) => ({
+        priority: 0,
+        shortTitle: msg[c.get('protocol') + 'Console'](),
+        icon: <Icon name='external-link' />,
+        id: `${idPrefix}-button-console-${c.get('protocol')}`,
+        confirmation: (
+          <ConsoleConfirmationModal consoleId={c.get('id')} vm={vm} />
+        ),
+      })).toJS()
+
+      if (vncConsole) {
+        consoles.push({
+          priority: 0,
+          shortTitle: msg.vncConsoleBrowser(),
+          actionDisabled: config.get('websocket') === null,
+          id: `${idPrefix}-button-console-browser`,
+          confirmation: (
+            <ConsoleConfirmationModal isNoVNC consoleId={vncConsole.get('id')} vm={vm} />
```

Comment:
Using `isNoVNC=true` for VNC is confusing. Perhaps we should re-name?
Additionally I think the intention here was probably to pass `isNoVnc` property not just `true`.
See [JSX syntax](https://reactjs.org/docs/jsx-in-depth.html#props-default-to-true)

Reply:
It should be removed, I forgot to remove it:
>You didn't read the value of the ClientModeVncDefault config val for deciding which VNC mode to use (native or noVNC) as we already discussed.

Thanks